redactevent.go

```diff
@@ -0,0 +1,131 @@
+package gomatrixserverlib
+
+import (
+	"encoding/json"
+)
+
+// rawJSON is a reimplementation of json.RawMessage that supports being used as
+// a value type
+type rawJSON []byte
+
+// MarshalJSON implements the json.Marshaller interface using a value receiver.
+// This means that rawJSON used as an embedded value will still encode correctly.
```

Comment:
You're going to need to give some usage examples here in order to convince me why you can't just use `json.RawMessage`. I *think* I know why (it incorrectly encodes as base 64) but given you're re-implementing parts of the stdlib, I think it warrants a better, more detailed explanation.